### PR TITLE
chore(all): add types

### PR DESCRIPTION
**Pull request by JeroenVinke**
Please check if I specified the optional/required parameters correctly.

I'm also not sure what the types of `evaluator` and `context`should be on `evaluateNavigationStrategy()`

### CODE REVIEW

File: src/router.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ export class Router {
     return false;
   }
 
-  registerViewPort(viewPort, name) {
+  registerViewPort(viewPort:any, name?:string) {
```
Comment (bryanrsmith) [inline src/router.js]: We'll eventually need an interface for view ports, but in the meantime `Object` would be more appropriate than `any`.

File: src/router.js [hunk 2]
```diff
@@ -46,7 +46,7 @@ export class Router {
     }
   }
 
-  configure(callbackOrConfig) {
+  configure(callbackOrConfig:any):Router {
```
Comment (bryanrsmith) [inline src/router.js]: `callbackOrConfig` is a `RouterConfiguration` or a callback that takes a `RouterConfiguration`.

File: src/router.js [hunk 3]
```diff
@@ -60,7 +60,7 @@ export class Router {
     return this;
   }
 
-  navigate(fragment, options) {
+  navigate(fragment?:string, options?:Object):boolean {
```
Comment (bryanrsmith) [inline src/router.js]: I don't believe `fragment` is optional.

File: src/router.js [hunk 4]
```diff
@@ -77,13 +77,13 @@ export class Router {
     this.history.navigateBack();
   }
 
-  createChild(container) {
+  createChild(container?:Router):Router {
```
Comment (bryanrsmith) [inline src/router.js]: `container` is a `Container` from aurelia/dependency-injection.

File: src/router.js [hunk 5]
```diff
@@ -77,13 +77,13 @@ export class Router {
...
     var childRouter = new Router(container || this.container.createChild(), this.history);
     childRouter.parent = this;
     return childRouter;
   }
 
-  createNavigationInstruction(url = '', parentInstruction = null) {
+  createNavigationInstruction(url:string = '', parentInstruction?:NavigationInstruction = null):Promise<any> {
```
Comment (bryanrsmith) [inline src/router.js]: This returns `Promise<NavigationInstruction>`

File: src/router.js [hunk 6]
```diff
@@ -133,12 +133,12 @@ export class Router {
...
   }
 
-  createNavigationContext(instruction) {
+  createNavigationContext(instruction:NavigationInstruction):NavigationContext {
     instruction.navigationContext = new NavigationContext(this, instruction);
     return instruction.navigationContext;
   }
 
-  generate(name, params) {
+  generate(name:string, params?:Object) {
```
Comment (bryanrsmith) [inline src/router.js]: This returns a string.

File: src/router.js [hunk 7]
```diff
@@ -147,7 +147,7 @@ export class Router {
     return createRootedPath(path, this.baseUrl, this.history._hasPushState); 
   }
 
-  createNavModel(config) {
+  createNavModel(config:NavModel):NavModel {
```
Comment (bryanrsmith) [inline src/router.js]: `config` is a route config here, which we'll need an interface for. In the meantime, `Object` would be better.

File: src/router.js [hunk 8]
```diff
@@ -215,15 +215,15 @@ export class Router {
...
     return !!(this.recognizer.hasRoute(name) || this.parent && this.parent.hasRoute(name));
   }
 
-  hasOwnRoute(name) {
+  hasOwnRoute(name:string):boolean {
     return this.recognizer.hasRoute(name);
   }
 
-  handleUnknownRoutes(config) {
+  handleUnknownRoutes(config?:string) {
```
Comment (bryanrsmith) [inline src/router.js]: `config` is a string, object, or function here.

File: src/router.js [hunk 9]
```diff
@@ -281,7 +281,7 @@ function validateRouteConfig(config) {
   }
 }
 
-function evaluateNavigationStrategy(instruction, evaluator, context){
+function evaluateNavigationStrategy(instruction:NavigationInstruction, evaluator, context): NavigationInstruction {
```
Comment (bryanrsmith) [inline src/router.js]: This returns a `Promise<NavigationInstruction>`

Comment (bryanrsmith) [inline src/router.js]: I believe `evaluator` is a function, and `context` is a route config object (`Object` is fine for now).

**Comment by bryanrsmith** [comment]
Thanks for working on this! I think there are just a few types to adjust, but it's looking good.

We don't update files in dist until we publish releases, so it'd be great if you could back out the changes there and only include the src files in your PR.

**Comment by JeroenVinke** [comment]
Made a new PR with these changes: https://github.com/aurelia/router/pull/145

**Pull request closed without merging**